MobilityData/mobility-feed-api
# feat: generate access token endpoint

**Summary:**
- Implemented the tokens API with a cloud function
- Setup basic structure to add more cloud functions in the future
- Add unit tests to the tokens' function.
- Refactor unit test and lint scripts, adding the new python functions.

Closes #136 

**Expected behavior:** 


**Testing tips:**

- Login to DEV environment and copy your refresh token
-  Execute the following curl command:
```
curl --location 'https://api-dev.mobilitydatabase.org/v1/tokens' \
--header 'Content-Type: application/json' \
--data '{
    "refresh_token": "HERE_YOUR_TOKEN"
}'
```
- Verify that you get a 200 HTTP status response with the following structure:
```
{
    "access_token": "ACCESS_TOKEN",
    "expiration_datetime_utc": "2023-11-13T21:46:35Z",
    "token_type": "Bearer"
}
```
- Use the newly generated access token to hit any Feeds API endpoint. Example:
```
curl --location 'https://api-dev.mobilitydatabase.org//v1/metadata' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer HERE_YOUR_ACCESS_TOKEN'
```

Please make sure these boxes are checked before submitting your pull request - thanks!
